sql/catalyst/src/main/scala/org/apache/spark/sql/types/ArrayType.scala

```diff
@@ -90,7 +90,7 @@ case class ArrayType(elementType: DataType, containsNull: Boolean) extends DataT
   override def sql: String = s"ARRAY<${elementType.sql}>"
 
   override private[spark] def asNullable: ArrayType =
-    ArrayType(elementType.asNullable, containsNull = true)
+    ArrayType(elementType.asNullable, containsNull)
```

Comment:
Hey, `asNullable` should return it as nullable. This is a general problem when you read it, it becomes nullable in SparkSQL file formats. We need to revisit nullability (given the community discussion so far) in general, not only specifically for this one.

Reply:
My bad , I agree `asNullable` should return it as nullable, it cannot be decided based on the current DF. for e.g below use case will fail if we don't return as `true`
`      val df = Seq(Some(Seq[Long](1L, 2L,3L)), None).toDF("seq")
      df.write.format("parquet").saveAsTable("testSchema");
      val df1 = Seq(Some(Seq(null)), None).toDF("seq")
      df1.write.format("parquet").mode("append").saveAsTable("testSchema")`

BTW what do you mean by We need to revisit nullability (given the community discussion so far) in general, not only specifically for this one.

I will close this PR and update this discussion in jira to close it as non issue

Reply:
For instance, when dataframe are loaded via file format sources like parquet, orc, etc., schema is forced to billable schema, which seems wrong. We should fix it across SparkSQL